vyper/parser/stmt.py

```diff
@@ -277,16 +280,39 @@ def parse_for(self):
 
         block_scope_id = id(self.stmt.orelse)
         self.context.start_blockscope(block_scope_id)
+
+        # type defaults to int128
+        typ_str = 'int128'
+        # goes through all keywords in iterator variable
+        # this loop can be used if new keyword arguments are added too
+        if len(self.stmt.iter.keywords) > 0:
+            for keyword in self.stmt.iter.keywords:
+                # set type equal to the value of the type keyword
+                if keyword.arg == 'type':
+                    typ_str = keyword.value.id
+
+        # handle invalid types
+        # this function raises an error for unsupported types
+        typ = canonicalize_type(BaseType(typ_str))
+
+        # was not working unless converted back to decimal
+        # should canonicalize_type be used if this is the case?
+        # or another mapping?
+        if typ == 'fixed168x10':
+            typ = 'decimal'
```

Comment:
Hmmm not sure canonicalize_type is really intended for this,  I would suggest one just creates a simple list of supported types for this functionality with a `if type not in (): raise TypeMismatch...`.

Reply:
We also have a `base_types` in `utils.py` that you could use for that list ;)